internal/modsdir/manifest.go

```diff
@@ -81,6 +81,13 @@ func ReadManifestSnapshot(r io.Reader) (Manifest, error) {
 				return nil, fmt.Errorf("invalid version %q for %s: %s", record.VersionStr, record.Key, err)
 			}
 		}
+
+		// Ensure Windows is using the proper modules directory format after
+		// reading the modules manifest
+		if string(filepath.Separator) != "/" {
+			record.Dir = filepath.FromSlash(record.Dir)
```

Comment:
Won't this have the same behavior/desired outcome if we had this statements without the `if`?

Reply:
That's a good point. The functions from `filepath` do have their own conditions that are essentially similar. I can remove the conditions here.